display/file.go

```diff
@@ -0,0 +1,113 @@
+package display
+
+import (
+	"fmt"
+	"io/ioutil"
+	"mime"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/makeworld-the-better-one/amfora/renderer"
+	"github.com/makeworld-the-better-one/amfora/structs"
+	"github.com/spf13/viper"
+)
+
+func pathFromURI(u string) string {
+	path := strings.TrimPrefix(u, "file://")
+	path = strings.TrimPrefix(path, "localhost") // localhost is a valid host for file URIs
+	path = strings.TrimPrefix(path, "/")         // Valid file URIs contains this additional slash
```

Comment:
This additional represents the top level directory, does it not? For example, a file URI could be `file:///usr/local/bin/foo`. If you remove the "additional slash", you will end up with a relative path instead of an absolute one: `usr/local/bin/foo`.

A simple test for this is to make sure running `amfora file:///` opens the top level directory successfully. The next test is to make sure things work okay on Windows without it. If they don't, then I'm fine with removing that slash only on windows, by checking `runtime.GOOS`.

Reply:
Once this is fixed and tested on Windows, I should be good to merge.

Reply:
Hmmm, I had some problems thinking about a solution that solves relative (to current working directory) paths with file URIs.
But apparently that is not supported in any form in a valid file URI

Also looking at Go's url.ParseRequestURI which handles your comments and all other edge cases, but don't allow relative URIs.

Would you be ok with a solution that just disallow this (and follows the standard)?
In that case, just swithcing to url.ParseRequestURI is probably the way to go

Reply:
> Would you be ok with a solution that just disallow this (and follows the standard)?

Yep! This is how I expected it to work.

> `url.ParseRequestURI`

Sounds good to me, better than what I was suggesting. Just return `.Path` after parsing. The only extra thing would be to handle the possibility of a parsing error, ie a malformed URL. Maybe that would happen in this func, maybe somewhere else, which would probably remove the need for this func entirely. Make sure `Error` is called so the user is aware.

And don't forget to test on Windows. Thanks!

Reply:
@makeworld-the-better-one: Ok, tested on windows as well
Seems to work as expected, eventhough I'm not a fan of how Windows paths works in general.

https://github.com/golang/go/issues/6027 has a discussion on how Go chose to implement this.
In short, there is no perfect way to deal with windows drive letters, but go/python/nodejs/java have settled on one of the solutions.

Reply:
Thanks for the link and testing! The Windows stuff is a bit weird, but as long as the file opens correctly.